src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -52,20 +52,20 @@
-            "            ddp_site.collaborator_id_prefix, req.bsp_collaborator_sample_id, req.ddp_participant_id, req.ddp_label, req.dsm_kit_request_id, req.bsp_collaborator_participant_id, " +
-            "            req.kit_type_id, req.external_order_status, req.external_order_number, req.external_order_date, req.external_response, kt.no_return, req.created_by, kit.kit_label FROM kit_type kt, ddp_kit_request req, ddp_kit kit, ddp_instance ddp_site " +
-            "            WHERE req.ddp_instance_id = ddp_site.ddp_instance_id AND req.kit_type_id = kt.kit_type_id AND kit.dsm_kit_request_id = req.dsm_kit_request_id " +
-            "            and kit.test_result is null " +
-            "            and not (kit.ups_tracking_status like 'D%' " +
-            "                    and kit.ups_return_status like 'D%' " +
-            "                    and kit.ups_tracking_status is not null " +
-            "                    and kit.ups_return_status is not null) " +
-            "            ) AS request " +
-            "            LEFT JOIN ddp_participant_exit ex ON (ex.ddp_instance_id = request.ddp_instance_id AND ex.ddp_participant_id = request.ddp_participant_id) " +
-            "            LEFT JOIN (SELECT subK.kit_type_id, subK.external_name from ddp_kit_request_settings dkc " +
-            "            LEFT JOIN sub_kits_settings subK ON (subK.ddp_kit_request_settings_id = dkc.ddp_kit_request_settings_id)) as subkits ON (subkits.kit_type_id = request.kit_type_id) " +
-            "            WHERE ex.ddp_participant_exit_id is null AND request.ddp_instance_id = ? AND NOT external_order_status <=> 'CANCELLED' AND external_order_number IS NOT NULL " ;
+    public static final String SQL_SELECT_EXTERNAL_KIT_NOT_DONE = "SELECT * FROM (SELECT kt.kit_type_name, ddp_site.instance_name, ddp_site.ddp_instance_id, ddp_site.base_url, ddp_site.auth0_token, ddp_site.migrated_ddp,\n" +
+            "                        ddp_site.collaborator_id_prefix, req.bsp_collaborator_sample_id, req.ddp_participant_id, req.ddp_label, req.dsm_kit_request_id, req.bsp_collaborator_participant_id,\n" +
+            "                        req.kit_type_id, req.external_order_status, req.external_order_number, req.external_order_date, req.external_response, kt.no_return, req.created_by, kit.kit_label FROM kit_type kt, ddp_kit_request req, ddp_kit kit, ddp_instance ddp_site\n" +
+            "                        WHERE req.ddp_instance_id = ddp_site.ddp_instance_id AND req.kit_type_id = kt.kit_type_id AND kit.dsm_kit_request_id = req.dsm_kit_request_id\n" +
+            "                        and kit.test_result is null and kit.CE_order is null\n" +
+            "                        and not (kit.ups_tracking_status is not null\n" +
```

Comment:
should we maybe change that to 
"kit.tracking_to_id is null or kit.tracking_return_id is null or kit.kit_label is null"
and loose the "not not" situation. because if we have all of these 3 ids, it means we already got the kit per confirmation and that the status must be shipped?

Reply:
Refining this a bit more--we don't need ups status at all.  We can just check kit label and tracking ids.